privacyidea/api/lib/prepolicy.py

```diff
@@ -995,8 +988,7 @@ def check_token_upload(request=None, action=None):
     :param filename:
     :return:
     """
-    params = request.all_data
-    upload_allowed = Match.admin(g, action=ACTION.IMPORT, realm=params.get("realm")).allowed()
+    upload_allowed = Match.admin(g, action=ACTION.IMPORT, user_obj=request.User).allowed()
```

Comment:
Also since we don't need the `user_obj` here we can remove the `user_obj` parameter from the `admin`-match function completely.

Reply:
We do not need the user object here because we might have none. And the tokens are only uploaded and assigned to a certain realm, but not to a certain user. Thus the request contains the realm parameter but not a complete user.
I will change this, to use the Match.generic here.

But we can need the user_obj in all other cases, because an admin e.g. is allowed to enroll a token **for** a certain **user**. And the user_obj is used to check exactly this with the policy definitions.

Reply:
I added a correct upload policy in 4913c72c

Reply:
Currently all uses of the `Match.admin()` function pass `user_obj=None`.

Reply:
I do not understand this. Please explain this to me in person.

Reply:
OK, now I see there are only three occurrences. I will keep the user_obj but do a default value=None.

Reply:
@plettich done in 5f746adf